Added support for 3D scatter plots.

landlockedsurfer opened the pull request:
Hi!
I've came across this [Stackoverflow entry](https://stackoverflow.com/questions/48366536/in-scala-is-it-possible-to-use-plotlys-scatter3d
) and decided to have a closer look on actually implementing 3D scatter plots.
This commit should add basic 3D scatter plot support, but lacks documentation and test cases. If you are interested I can add those also.
Best regards,
Manfred

pbugnion commented:
Thanks a lot for submitting this! It looks really great. 

I'll test it out properly in the next couple of days.

pbugnion commented:
This looks great! For anyone wanting to try this out:

```scala
import co.theasi.plotly._

object Scatter3DExample extends App {
  val plot = ThreeDPlot().withScatter3D(List(1.0, 2.0), List(4.0, 5.0), List(-1.0, -3.0), ScatterOptions())
  println(draw(plot, "example-3d-112"))
}
```

... produces [this](https://plot.ly/~pbugnion/809/).

A couple of thoughts:

 - It would be great to have default `ScatterOptions` for `.withScatter3D(xs, ys, zs)` -- for instance, the definition for `withScatter` is:

```scala
  def withScatter[X: Writable, Y: Writable](
      xs: Iterable[X],
      ys: Iterable[Y],
      options: ScatterOptions = ScatterOptions()
  ): CartesianPlot = // ...
```

- At least one 'integration test' -- in `WriterSpec`, we generate a full plot, dispatch it to plotly and then query the plotly API to check the plot is well-formed. It should be fairly easy to adapt the existing 3D test.

- A minimal scaladoc for `withScatter3D` (e.g. mirroring one of the `withSurface` scaladocs).

Thanks again for putting this together. It's a great addition. I'll cut a new release when this is merged.

landlockedsurfer commented:
Thanks for your motivating feedback!
I've now added an additional commit to at least rudimentarily cover the points you mentioned.
Best regards,
Manfred

pbugnion commented on src/main/scala/co/theasi/plotly/Plot.scala:
```diff
@@ -263,6 +263,15 @@ object CartesianPlot {
+  * val xs = Vector(1.0, 2.0)
+  * val ys = Vector(4.0, 5.0)
+  * val zs = Vector(-1.0, -3.0)
+  * val plot = ThreeDPlot().withScatter3D(xs, ys, zs, ScatterOptions())
```
If we are using the default, we probably don't need to specify `ScatterOptions` explicitly in the example, to keep it as simple as possible? 

People who want to know more about the options can look at the API docs for `withScatter3D`?

pbugnion commented:
This is great! Thanks for writing tests and Scaladoc. I made a tiny comment inline.

I was also wondering if we should rename the method to `withScatter`, so it mirrors the `CartesianPlot` structure method name. Having `ThreeDPlot().withScatter3D` feels redundant. What do you think?

Once that's resolved, I think this is good to be merged.

landlockedsurfer commented:
Thanks for the review. Those are good points.
I've refactored `withScatter3D` to  `withScatter` and also dropped the explicit `ScatterOptions` from the documentation.

pbugnion commented:
Thanks! I'll give this a final run-through tomorrow morning. Thanks a lot for taking the time to push this through, it's a really valuable addition.
